full-stack-tests/presentation/src/learning-snippets/Content.test.ts

```diff
@@ -4,589 +4,1012 @@
       });
-      expect(content!.descriptor.fields).to.containSubset([{
-        label: "Category",
-        category: { label: "Custom Category" },
-      }, {
-        label: "Code",
-        category: { label: "Custom Category" },
-      }, {
-        label: "Model",
-        category: { label: "Custom Category" },
-      }, {
-        label: "User Label",
-        category: { label: "Custom Category" },
-      }]);
-    });
 
-    it("uses `propertyOverrides` attribute", async () => {
-      // __PUBLISH_EXTRACT_START__ ContentModifier.PropertyOverrides.Ruleset
-      // The ruleset has a content rule that returns content of given input instances. There's also
-      // a content modifier that customizes display of `bis.GeometricElement3d` properties.
-      const ruleset: Ruleset = {
-        id: "example",
-        rules: [{
-          ruleType: RuleTypes.Content,
-          specifications: [{
-            // load content for given input instances
-            specType: ContentSpecificationTypes.SelectedNodeInstances,
+      it("uses `relatedProperties` attribute", async () => {
+        // __PUBLISH_EXTRACT_START__ ContentInstancesOfSpecificClasses.RelatedProperties.Ruleset
+        // The specification returns content for `bis.SpatialViewDefinition` and
+        // related `bis.DisplayStyle` properties.
+        const ruleset: Ruleset = {
+          id: "example",
+          rules: [{
+            ruleType: RuleTypes.Content,
+            specifications: [{
+              specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
+              classes: { schemaName: "BisCore", classNames: ["SpatialViewDefinition"] },
+              relatedProperties: [{
+                propertiesSource: {
+                  relationship: { schemaName: "BisCore", className: "ViewDefinitionUsesDisplayStyle" },
+                  direction: RelationshipDirection.Forward,
+                },
+              }],
+            }],
           }],
+        };
+        // __PUBLISH_EXTRACT_END__
+
+        // Ensure that derived `bis.DisplayStyle` instance properties are also returned with `bis.SpatialViewDefinition` content.
+        const content = await Presentation.presentation.getContent({
+          imodel,
+          rulesetOrId: ruleset,
+          keys: new KeySet(),
+          descriptor: {},
+        });
+
+        expect(content!.contentSet.length).to.eq(4);
+        expect(content!.descriptor.fields).to.containSubset([
+          { label: "3D Display Style" },
+        ]).and.to.have.lengthOf(18);
+        expect((content!.descriptor.fields.find((record) => record.isNestedContentField()) as NestedContentField)!.nestedFields).to.containSubset([
+          { name: "pc_bis_Element_Model_2" },
+          { name: "pc_bis_Element_CodeValue_2" },
+          { name: "pc_bis_Element_UserLabel_2" },
+          { name: "pc_bis_DefinitionElement_IsPrivate_2" },
+        ]);
```

Comment:
Why not check nested fields using the above `containSubset` call? Also, I'd check them using display labels rather than names - the names occasionally change based on code changes.